### PR TITLE
Fix undefined `PATH_MAX`

### LINKED ISSUE #5614: widelands 1.1-rc1: compilation error: use of undeclared identifier 'PATH_MAX'
**Describe the bug**
When compiling widelands 1.1-rc1 on OpenMandriva Cooker I see build error:

```
/builddir/build/BUILD/widelands-1.1-rc1/src/io/filesystem/filesystem.cc:183:11: error: use of undeclared identifier 'PATH_MAX'
DEBUG util.py:446:          char cwd[PATH_MAX + 1];
DEBUG util.py:446:                   ^
DEBUG util.py:446:  /builddir/build/BUILD/widelands-1.1-rc1/src/io/filesystem/filesystem.cc:188:35: error: use of undeclared identifier 'PATH_MAX'
DEBUG util.py:446:          char* const result = getcwd(cwd, PATH_MAX);
DEBUG util.py:446:                                           ^
DEBUG util.py:446:  2 errors generated.
```

Same error on Clang 15.0.2 and GCC compiler.

Full build log: https://file-store.openmandriva.org/api/v1/file_stores/9e9dcb617964e9d37469c17b9007b6fb930fab0d.log?show=true

**To Reproduce**
1. Compile from source widelands 1.1-rc1
2. See error

**Expected behavior**
Compilation ends without errors

**Version:**
 - OS: [e.g. OpenMandriva Cooker]
 - Widelands Version: 1.1-rc1]

### PR DESCRIPTION
**Pull request by Noordfrees**
Quick fix #5614

@AngryPenguinPL Could you please test if this patch works for you?

### CODE REVIEW

File: src/io/filesystem/filesystem.cc [hunk 1]
```diff
@@ -59,6 +59,10 @@
 #define PATH_MAX MAX_PATH
 #endif
 
+#ifndef PATH_MAX
+#define PATH_MAX 0x1000000
```
Comment (klaus-halfmann) [inline src/io/filesystem/filesystem.cc]: Where did you get this value from?

**Review by klaus-halfmann — COMMENTED** [review]
Fix is obvious, but
* Please add a comment referring to the original Bug
* Please add comment where this 0x1000000 value comes from.

I found `#define PATH_MAX                 1024   /* max bytes in pathname */`
in some macOS syslimits.h whih would be just 0x400 ?

**Comment by AngryPenguinPL** [comment]
@Noordfrees Yes, I can confirm that this patch works/

Thanks.

**Review by klaus-halfmann — APPROVED** [review]
Sorry  wanted to approve this already ths morning, lets go!

**Pull request merged**